lib/UI/calendar/widgets/date_widget.dart

```diff
@@ -3,96 +3,97 @@ import 'package:exerlog/Bloc/workout_bloc.dart';
+            /// This results in bad state, due to parsing error
+            // Log.error(e.toString(), stackTrace: stackTrace);
+          }
+          return GestureDetector(
+            onTap: () {
+              if (workout != null)
+                _navigateToPreviousWorkoutScreen(
+                  context,
+                  workout,
+                );
+            },
+            child: _buildDayContainer(hasWorkout: workout != null),
           );
         } else {
-          if (snapshot.hasError) {
-            return Container(
-              padding: EdgeInsets.only(top: 5),
-              height: 40,
-              width: 35,
-              child: Center(
-                  child: Text(
-                widget.date.day.toString(),
-                style: whiteTextStyleSmall,
-              )),
-            );
-          } else if (snapshot.hasData) {
-            try {
-              Workout workout = Workout.fromJsonQuerySnapshot(snapshot.data!);
-              workout.id = snapshot.data!.docs.last.id;
-
-              return Container(
-                padding: EdgeInsets.only(top: 5),
-                height: 40,
-                width: 35,
-                child: RaisedGradientButton(
-                  onPressed: () {
-                    Navigator.of(context).push(
-                      MaterialPageRoute(
-                          builder: (context) => PrevWorkoutPage(workout)),
-                    );
-                  },
-                  radius: 10,
-                  gradient: LinearGradient(
-                    colors: <Color>[Color(0xFF34D1C2), Color(0xFF31A6DC)],
-                  ),
-                  child: Text(
-                    widget.date.day.toString(),
-                    style: buttonTextSmall,
-                  ),
-                ),
-              );
-            } catch (Exception) {
-              return Container(
-                padding: EdgeInsets.only(top: 5),
-                height: 40,
-                width: 35,
-                child: Center(
-                    child: Text(
-                  widget.date.day.toString(),
-                  style: whiteTextStyleSmall,
-                )),
-              );
-            }
-          } else {
-            return Container(
-              padding: EdgeInsets.only(top: 5),
-              height: 40,
-              width: 35,
-              child: Center(
-                  child: Text(
-                widget.date.day.toString(),
-                style: whiteTextStyleSmall,
-              )),
-            );
-          }
+          return _buildDayContainer();
         }
       },
     );
   }
+
+  Container _buildDayContainer({bool hasWorkout = false}) {
+    return Container(
+      margin: EdgeInsets.all(5),
+      padding: EdgeInsets.all(10),
+      decoration: hasWorkout ? _hasWorkoutDecoration() : null,
+      child: Center(
+        child: Text(
+          date.day.toString(),
+          textScaleFactor: 1,
```

Comment:
hmm why does this have a textScaleFactor ?

Reply:
So that if user increases device font size from the setting this doesn't break due to overflow

Reply:
gotcha